### PR TITLE
catgirl: init at 1.6

### PR DESCRIPTION
**Pull request by xfnw**
###### Motivation for this change
package the catgirl irc client

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xfnw** [comment]
oops, forgot to add myself to the maintainers list

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{lib, stdenv, fetchurl, ncurses, libressl, man, pkg-config, ctags}:
+
+stdenv.mkDerivation rec {
+  name = "catgirl-1.6";
```
Comment (fabaff) [inline pkgs/applications/networking/irc/catgirl/default.nix:4]:
```suggestion
  pname = "catgirl";
  version = "1.6";
```

see https://nixos.org/manual/nixpkgs/unstable/#sec-using-stdenv ([RFC 0035](https://github.com/NixOS/rfcs/pull/35))

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  src = fetchurl {
+    url = "https://git.causal.agency/catgirl/snapshot/${name}.tar.gz";
```
Comment (fabaff) [inline pkgs/applications/networking/irc/catgirl/default.nix:7]: With the change from above

```suggestion
    url = "https://git.causal.agency/catgirl/snapshot/${pname}-$version{}.tar.gz";
```

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    sha256 = "0shg02zidqqmvywqqsaazlgg9rd5lhhrvjx6n0lzmdfaawxywciv";
+  };
+
+  nativeBuildInputs = [ pkg-config ncurses libressl man ctags ];
+
+  meta = with lib; {
+    homepage = "https://git.causal.agency/catgirl/about/";
+    license = lib.licenses.gpl3;
```
Comment (fabaff) [inline pkgs/applications/networking/irc/catgirl/default.nix:15]: See line 13 (`with lib`)

```suggestion
    license = licenses.gpl3Plus;
```

The header in the source file says:

```text
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 ```

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    description = "A TLS-only terminal IRC client";
+    platforms = lib.platforms.unix;
```
Comment (fabaff) [inline pkgs/applications/networking/irc/catgirl/default.nix:17]: See line 13 (`with lib`)

```suggestion
    platforms = platforms.unix;
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 115159` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catgirl</li>
  </ul>
</details>

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+stdenv.mkDerivation rec {
+  pname = "catgirl";
+  version = "1.6";
+
+  src = fetchurl {
+    url = "https://git.causal.agency/catgirl/snapshot/${pname}-${version}.tar.gz";
+    sha256 = "0shg02zidqqmvywqqsaazlgg9rd5lhhrvjx6n0lzmdfaawxywciv";
+  };
+
+  nativeBuildInputs = [ pkg-config ncurses libressl man ctags ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/irc/catgirl/default.nix:12]:
```suggestion
  nativeBuildInputs = [ pkg-config man ];
  
  buildInputs = [ ncurses libressl ctags ];
  
  strictDeps = true;
```

Most likely this is correct but I am not completely sure.

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+  meta = with lib; {
+    homepage = "https://git.causal.agency/catgirl/about/";
+    license = licenses.gpl3Plus;
+    description = "A TLS-only terminal IRC client";
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ xfnw ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/irc/catgirl/default.nix:22]: One final new line to much.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
please squash the ``catgirl: init at 1.6`` commits.

File: pkgs/applications/networking/irc/catgirl/default.nix (hunk 5 above)
Comment (xfnw) [reply pkgs/applications/networking/irc/catgirl/default.nix:12]: seems `ctags` still needs to be in the `nativeBuildInputs`

**Comment by zseri** [comment]
It might be a good idea to sort the package lists (e.g. argument list in the header, `nativeBuildInputs`, `buildInputs`) alphabetically.

**Comment by xfnw** [comment]
> It might be a good idea to sort the package lists (e.g. argument list in the header, `nativeBuildInputs`, `buildInputs`) alphabetically.

@zseri hmm, i can't seem to find any other packages that sort the build inputs, is that a normal thing to do?

**Comment by zseri** [comment]
The last few PRs that I reviewed all had sorted build inputs, and I think it is a normal thing to do.
This does not apply to the argument list in the header: it is often not "simply" sorted, and there exist multiple conflicting conventions applied to different packages in nixpkgs.

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  pname = "catgirl";
+  version = "1.6";
+
+  src = fetchurl {
+    url = "https://git.causal.agency/catgirl/snapshot/${pname}-${version}.tar.gz";
+    sha256 = "0shg02zidqqmvywqqsaazlgg9rd5lhhrvjx6n0lzmdfaawxywciv";
+  };
+
+  nativeBuildInputs = [ pkg-config ctags ];
+  buildInputs = [ ncurses libressl man ];
```
Comment (zseri) [inline pkgs/applications/networking/irc/catgirl/default.nix:13]:
```suggestion
  nativeBuildInputs = [ ctags pkg-config ];
  buildInputs = [ libressl man ncurses ];
```

Comment (xfnw) [reply pkgs/applications/networking/irc/catgirl/default.nix:13]: oh that makes a lot more sense, i thought you were talking about putting the entire lines in alphabetical order

**Comment by xfnw** [comment]
@SuperSandro2000 could you re-review?

File: pkgs/applications/networking/irc/catgirl/default.nix [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{lib, stdenv, fetchurl, ncurses, libressl, man, pkg-config, ctags}:
```
Comment (zseri) [inline pkgs/applications/networking/irc/catgirl/default.nix:1]: (optional)
```suggestion
{ ctags, fetchurl, lib, libressl, man, ncurses, pkg-config, stdenv }:
```

**Review by fogti — APPROVED** [review]
builds successfully with Nix (2.3.10) on Gentoo.

File: maintainers/maintainer-list.nix [hunk 9]
```diff
@@ -10268,6 +10268,12 @@
     githubId = 36407913;
     name = "Uli Baum";
   };
+  xfnw = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:10271]: Please put this into a separate commit with the message ``maintainers: add xfnw``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
other than that LGTM

File: maintainers/maintainer-list.nix (hunk 9 above)
Comment (xfnw) [reply maintainers/maintainer-list.nix:10271]: done
